csrc/evaluator_common.cpp

```diff
@@ -190,14 +190,10 @@ void PrecomputedValues::initializeValueList(
     // Use an expression evaluator to test if value is const
     if (sorted_value_list[i]->isConstScalar()) {
       is_constant_[i] = true;
-      if (sorted_value_list[i]->isIntegralScalar()) {
-        values_[i] = PolymorphicValue(sorted_value_list[i]->evaluateInt());
-      }
-      if (sorted_value_list[i]->isFloatingPointScalar()) {
-        values_[i] = PolymorphicValue(sorted_value_list[i]->evaluateDouble());
-      }
-      if (sorted_value_list[i]->isABool()) {
-        values_[i] = PolymorphicValue(sorted_value_list[i]->evaluateBool());
+      if (sorted_value_list[i]->isIntegralScalar() ||
+          sorted_value_list[i]->isFloatingPointScalar() ||
+          sorted_value_list[i]->isABool()) {
+        values_[i] = PolymorphicValue(sorted_value_list[i]->evaluate());
       }
```

Comment:
I think we can just do
```suggestion
      values_[i] = sorted_value_list[i]->evaluate();
```
In the past, `evaluate` only supports these three dtypes, that's why we have that check. But this is no longer the case, so we should just remove these checks.

Reply:
Thanks! Adressed.